ocaml/xapi/xapi_host.ml

```diff
@@ -3054,21 +3059,35 @@ let try_restart_device_models_for_recommended_guidances ~__context ~host =
        * device models are running for this VM *)
       None
 
-let apply_recommended_guidances ~__context ~self =
+let apply_recommended_guidances ~__context ~host =
+  (* This function runs on master host *)
+  Helpers.assert_we_are_master ~__context ;
   try
     let open Updateinfo in
-    Db.Host.get_recommended_guidances ~__context ~self |> function
+    Db.Host.get_recommended_guidances ~__context ~self:host |> function
     | [] ->
-        try_restart_device_models_for_recommended_guidances ~__context
-          ~host:self
+        try_restart_device_models_for_recommended_guidances ~__context ~host
     | [`reboot_host] ->
-        reboot ~__context ~host:self
+        let local_fn = reboot ~host in
+        Xapi_host_helpers.with_host_operation ~__context ~self:host
+          ~doc:"Host.reboot" ~op:`reboot (fun () ->
+            Message_forwarding.do_op_on ~local_fn ~__context ~host
+              (fun session_id rpc ->
+                Client.Client.Host.reboot ~rpc ~session_id ~host
+            )
+        )
     | [`restart_toolstack] ->
-        try_restart_device_models_for_recommended_guidances ~__context
-          ~host:self ;
-        restart_agent ~__context ~host:self
+        try_restart_device_models_for_recommended_guidances ~__context ~host ;
+        let local_fn = restart_agent ~host in
+        Xapi_host_helpers.with_host_operation ~__context ~self:host
+          ~doc:"Host.reboot" ~op:`reboot (fun () ->
```

Comment:
Why the doc host be `Host.reboot`?

Reply:
Removed "Xapi_host_helpers.with_host_operation" here, as it is not used in "Message_forwarding.restart_agent"